lib/mix/tasks/g.ex

```diff
@@ -9,25 +9,54 @@ defmodule Mix.Tasks.Arc do
     @moduledoc """
     A task for generating arc uploader modules.
 
-    The generated attachment definition is stored in `web/uploaders`.
+    If generating an uploader in a Phoenix project, your a uploader will be generated in
+    lib/[APP_NAME]_web/uploaders/
 
     ## Example
 
-        mix arc.g avatar   # creates web/uploaders/avatar.ex
+        mix arc.g avatar # creates  lib/[APP_NAME]_web/uploaders/avatar.ex
+
+
+    If not generating an uploader in a Phoenix project, then you must pass the path to where the
+    uploader should be generated.
+
+    ## Example
+
+        mix arc.g avatar  uploaders/avatar.ex # creates uploaders/avatar.ex
     """
 
     def run([model_name]) do
+      app_name = Mix.Project.config[:app]
+      if (File.exists?("lib/#{app_name}_web/")) do
+        project_module_name = camelize(to_string(app_name))
+        generate_phx_uploader_file(model_name, project_module_name)
+      else
+        raise "path must be passed when generating an uploader."
+      end
+    end
+
+    def run([model_name, path]) do
       app_name = Mix.Project.config[:app]
       project_module_name = camelize(to_string(app_name))
-      generate_uploader_file(model_name, project_module_name)
+      generate_uploader_file(model_name, project_module_name, path)
     end
 
     def run(_) do
       IO.puts "Incorrect syntax. Please try mix arc.g <model_name>"
     end
 
-    defp generate_uploader_file(model_name, project_module_name) do
-      model_destination = Path.join(System.cwd(), "/web/uploaders/#{underscore(model_name)}.ex")
+    defp generate_uploader_file(model_name, project_module_name, path) do
+      model_destination = Path.join(System.cwd(), "#{path}/#{underscore(model_name)}.ex")
+      create_uploader(model_name, project_module_name, model_destination)
+    end
+
+    defp generate_phx_uploader_file(model_name, project_module_name) do
```

Comment:
I think these functions could be simplified into one, since they have different arity anyways

```elixir
defp generate_uploader_file(model_name, project_module_name) do
  generate_uploader_file(model_name, project_module_name, "/lib/#{ Mix.Project.config[:app] }/uploaders")
end

defp generate_uploader_file(model_name, project_module_name, path) do
  model_destination = Path.join(System.cwd(), "#{path}/#{underscore(model_name)}.ex")
  create_uploader(model_name, project_module_name, model_destination)
end
```

Alternatively, the `run([model_name])` method could just call the `generate_uploader_file` directly, and then the 

```elixir
   def run([model_name]) do
      app_name = Mix.Project.config[:app]
      if (File.exists?("lib/#{app_name}_web/")) do
        project_module_name = camelize(to_string(app_name))
        generate_uploader_file(model_name, project_module_name, Mix.Project.config[:app])
      else
        raise "path must be passed when generating an uploader."
      end
    end
```

Reply:
Personal preference is to have the code be as explicit as possible about what it's doing. Function names are one of the mechanisms I personally use to achieve that. 

Having the dedicated functions & function names rather than relying on arity and implied rules seems like it would be easier to grok and maintain.

Reply:
I see your point, i guess i will leave the matter up to mr @stavro. Personally, i think i might have been scarred by the ruby-shop im working at. Some of our method names go on for miles.

I usually prefer to leave the code be as lean as possible, and explain potentially confusing parts with comments and docs